### DISCUSSION TITLE
How to mock a dynamically imported standalone component inside a standalone component

### DISCUSSION THREAD

**Discussion by AE1NS** [help-me-mom/ng-mocks#4693] [Q&A] [answered]
Hey,

I have two standalone components, Test1Component and Test2Component. Test1Component dynamically imports Test2Component `import(Test2Component).then(...`. Is there a way to get a mocked version of Test2Component?

```
beforeEach(() => {
    return MockBuilder(Test1Component).mock(Test2Component, { shallow: true });
});
```
as the docs says, this mocks only the imports and thats working as expected, but how can I mock the component itself and ignore its imports? This also does not work and also seem to just mock the imports but everytime, my implementation is called (tested it with a console.log):
```
beforeEach(() => {
    return MockBuilder(Test1Component).mock(Test2Component, { shallow: false });
});
```

**Comment by satanTime** [comment, marked as answer]
Updated

14.6.0 supports that: https://ng-mocks.sudo.eu/guides/mock/dynamic-components


Original

Hi @AE1NS, please submit a feature request with an example of the code and how you would like to mock it.

You can use https://codesandbox.io/s/github/help-me-mom/ng-mocks-sandbox?file=/src/test.spec.ts as a template.

**Reply by AE1NS** [reply]
I tried to create an example here https://codesandbox.io/s/sweet-brook-vtw748?file=/src/test.spec.ts but unfortunately I cant get it run to load the component dynamically and get an error output. In my implementation this works fine. But the question is, if its possible to mock a component that has been dynamically imported with the help of the MockBuilder.

**Reply by satanTime** [reply]
Thanks for sharing. I reproduced on my side and looking for a way to intercept that.

**Reply by satanTime** [reply]
Hi there, 14.6.0 supports this now: https://github.com/help-me-mom/ng-mocks/releases/tag/v14.6.0.

Docs are here: https://ng-mocks.sudo.eu/guides/mock/dynamic-components
